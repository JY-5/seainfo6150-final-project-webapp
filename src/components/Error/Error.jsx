import React from 'react'
import { useHistory } from "react-router-dom";
import styles from "./Error.module.css";

const Error = () => {
    const history = useHistory();
    return (
        <div className={styles.container}>
            <div>
                <h1>Whoops!</h1>
                <p>We couldn't find the page you are looking for</p>        
                <button onClick={() => history.push('/') } >Go home</button>
            </div>
            <img alt="a bitten donut" src="https://webstockreview.net/images/donut-clipart-bitten-donut-2.jpg"/>
        </div>
    )
}

export default Error
