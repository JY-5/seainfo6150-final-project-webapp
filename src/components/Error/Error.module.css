.container {
    display: flex;
    flex-flow: row wrap;
    font-family: 'Prompt', sans-serif;
    justify-content: space-around;
    margin-top: 2rem;
    margin-left: 2rem;
    margin-right: 2rem;
}

.container h1 {
    color: #539222;
}

.container p {
    font-size: 1rem;
}

.container button {
    margin: 1rem 0;
    padding: 0.5rem;
}

.container img {
    width: 20rem;
    height: 20rem;
    margin-top: 2rem;
    max-width: 100%; 
    height: auto;
}