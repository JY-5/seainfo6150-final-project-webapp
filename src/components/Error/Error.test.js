import React from "react";
import Error from './Error';
import { render } from "@testing-library/react";

describe("Add Error tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Error/>);
      expect(container).toMatchSnapshot();
    });
});