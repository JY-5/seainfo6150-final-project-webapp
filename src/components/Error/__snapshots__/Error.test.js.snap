// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Error tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <h1>
        Whoops!
      </h1>
      <p>
        We couldn't find the page you are looking for
      </p>
      <button>
        Go home
      </button>
    </div>
    <img
      alt="a bitten donut"
      src="https://webstockreview.net/images/donut-clipart-bitten-donut-2.jpg"
    />
  </div>
</div>
`;
